packages/phone-number/lib/parse/index.js

```diff
@@ -19,56 +20,49 @@ export default function parse({ input, countryCode = 'US' }) {
     extension: null,
 
   };
-  const cleanInput = (input || '').replace(cleanRegex, '');
-  const startWithPlus = cleanInput[0] === '+';
-  const withoutPlus = cleanInput.replace(plusRegex, '');
-  const startWithStar = withoutPlus[0] === '*';
-  if (startWithPlus && startWithStar) {
+
+  var phoneNumberIsValid = phoneNumber => phoneNumber.length > 3 && validRegExp.test(phoneNumber);
+
+  var isServiceNumber = tokens => !tokens[0] && tokens[1];
```

Comment:
Extension delimeter can be '*' or '#', but service numbers can only start with '*'. By checking the tokens after splitting using extensionDelimeter does not ensure that the first character is a '*'.